src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/CaseEventHandlerLivenessHealthControllerTest.java

```diff
@@ -0,0 +1,126 @@
+        // GIVEN
+        when(caseEventMessageCacheService.getAllMessagesInNewState("validEnvironment"))
+            .thenReturn(Collections.emptyList());
+
+        // WHEN
+        Health health = caseEventHandlerLivenessHealthController.health();
+
+        // THEN
+        assertEquals(UP, health.getStatus());
+    }
+
+    @Test
+    void test_get_state_for_success_with_messages_over_threshold_and_still_processing() {
+
+        List<CaseEventMessageEntity> messages = new ArrayList<>();
+
+        IntStream.range(1,105).forEach(x -> {
+            CaseEventMessageEntity caseEventMessage = new CaseEventMessageEntity();
+            caseEventMessage.setSequence((long) x);
+            caseEventMessage.setCaseId("1234");
+            caseEventMessage.setMessageId("ABCD");
+            caseEventMessage.setState(MessageState.NEW);
+            caseEventMessage.setReceived(LocalDateTime.now().minusMinutes(45));
+            caseEventMessage.setEventTimestamp(LocalDateTime.now().minusMinutes(50));
+            messages.add(caseEventMessage);
+        });
+
+        // GIVEN
+        when(caseEventMessageCacheService.getAllMessagesInNewState("validEnvironment")).thenReturn(messages);
+
+        // WHEN
+        Health health = caseEventHandlerLivenessHealthController.health();
+
+        // THEN
+        assertEquals(UP, health.getStatus());
+    }
+
+    @Test
+    void test_get_state_for_success_if_messages_check_disabled_in_current_environment() {
+        // GIVEN
+        setField(caseEventHandlerLivenessHealthController, "environment", "invalidEnvironment");
+        setField(caseEventHandlerLivenessHealthController, "newMessageLivenessStateCheckEnvEnabled",
+                 "validEnvironment,test");
+
+        // WHEN
+        Health health = caseEventHandlerLivenessHealthController.health();
+
+        // THEN
+        assertEquals(UP, health.getStatus());
+        verifyNoInteractions(caseEventMessageCacheService);
+    }
+
+    @Test
+    void test_get_state_for_failure() {
+
+        List<CaseEventMessageEntity> messages = new ArrayList<>();
+
+        IntStream.range(1,105).forEach(x -> {
+            CaseEventMessageEntity caseEventMessage = new CaseEventMessageEntity();
+            caseEventMessage.setSequence((long) x);
+            caseEventMessage.setCaseId("1234");
+            caseEventMessage.setMessageId("ABCD");
+            caseEventMessage.setState(MessageState.NEW);
+            caseEventMessage.setReceived(LocalDateTime.now().minusMinutes(90));
+            caseEventMessage.setEventTimestamp(LocalDateTime.now().minusHours(2L));
+            messages.add(caseEventMessage);
+        });
+
+        // GIVEN
+        when(caseEventMessageCacheService.getAllMessagesInNewState("validEnvironment")).thenReturn(messages);
+
+        // WHEN
+        Health health = caseEventHandlerLivenessHealthController.health();
+
+        // THEN
+        assertEquals(DOWN, health.getStatus());
+    }
```

Comment:
Prob do with a scenario where the DB call returns an exception/error.

Reply:
Done now.